src/authorization/WebAclReader.ts

```diff
@@ -61,55 +63,54 @@ export class WebAclAuthorizer extends Authorizer {
    * Will throw an error if this is not the case.
    * @param input - Relevant data needed to check if access can be granted.
    */
-  public async handle({ identifier, permissions, credentials }: AuthorizerArgs): Promise<WebAclAuthorization> {
+  public async handle({ identifier, credentials }: PermissionReaderInput):
+  Promise<PermissionSet> {
     // Determine the required access modes
-    const modes = (Object.keys(permissions) as (keyof PermissionSet)[]).filter((key): boolean => permissions[key]);
-    this.logger.debug(`Checking if ${credentials.webId} has ${modes.join()} permissions for ${identifier.path}`);
-
-    // Determine the full authorization for the agent granted by the applicable ACL
-    const acl = await this.getAclRecursive(identifier);
-    const authorization = await this.createAuthorization(credentials, acl);
+    this.logger.debug(`Checking permissions of ${credentials.agent?.webId} for ${identifier.path}`);
 
-    // Verify that the authorization allows all required modes
-    for (const mode of modes) {
-      this.requirePermission(credentials, authorization, mode);
-    }
-    this.logger.debug(`${credentials.webId} has ${modes.join()} permissions for ${identifier.path}`);
-    return authorization;
-  }
+    const isAcl = this.aclStrategy.isAuxiliaryIdentifier(identifier);
+    const mainIdentifier = isAcl ? this.aclStrategy.getAssociatedIdentifier(identifier) : identifier;
 
-  /**
-   * Checks whether the agent is authenticated (logged in) or not (public/anonymous).
-   * @param agent - Agent whose credentials will be checked.
-   */
-  private isAuthenticated(agent: Credentials): agent is ({ webId: string }) {
-    return typeof agent.webId === 'string';
+    // Determine the full authorization for the agent granted by the applicable ACL
+    const acl = await this.getAclRecursive(mainIdentifier);
+    return this.createPermissions(credentials, acl, isAcl);
   }
 
   /**
    * Creates an Authorization object based on the quads found in the ACL.
-   * @param agent - Agent whose credentials will be used for the `user` field.
+   * @param credentials - Credentials to check permissions for.
    * @param acl - Store containing all relevant authorization triples.
+   * @param isAcl - If the target resource is an acl document.
    */
-  private async createAuthorization(agent: Credentials, acl: Store): Promise<WebAclAuthorization> {
-    const publicPermissions = await this.determinePermissions({}, acl);
-    const agentPermissions = await this.determinePermissions(agent, acl);
+  private async createPermissions(credentials: CredentialSet, acl: Store, isAcl: boolean):
+  Promise<PermissionSet> {
+    const publicPermissions = await this.determinePermissions(acl, credentials.everyone);
+    const agentPermissions = await this.determinePermissions(acl, credentials.agent);
+
+    // Agent at least has the public permissions
```

Comment:
It was written like this to stay consistent with the WebACL behaviour we already had: the authorization object it used to output was also constructed in such a way that the agent permissions were always at least the public permissions. One reason for this is that this is required for the WAC-Allow headers, but we could also catch this behaviour in the WAC-Allow handler actually, so this is not really relevant there.

Besides that we should look at how acl behaviour is supposed to work. The twitter example could indeed be a case where we want to differentiate between public and agent, but this is not behaviour that you can define in acl. If the acl document contains an authorization with `acl:agentClass foaf:Agent`, acl requires a logged in agent to also have those permissions.

Reply:
I do think that perhaps this can be removed because the access-checkers are currently written in such a way that the agent will have permissions in these cases. The only other reason for this then would be the wac-allow headers but that can be moved to the relevant class.

Reply:
> the agent permissions were always at least the public permissions. One reason for this is that this is required for the WAC-Allow header

Ah that's a requirement? Would perhaps handle it there then indeed.

> Besides that we should look at how acl behaviour is supposed to work. The twitter example could indeed be a case where we want to differentiate between public and agent, but this is not behaviour that you can define in acl.

ACL behavior is currently monotonic; this is more about the genericness of components.

> I do think that perhaps this can be removed because the access-checkers are currently written in such a way that the agent will have permissions in these cases. The only other reason for this then would be the wac-allow headers but that can be moved to the relevant class.

+1